sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -599,8 +599,8 @@ class Analyzer(override val catalogManager: CatalogManager)
       val aggForResolving = h.child match {
         // For CUBE/ROLLUP expressions, to avoid resolving repeatedly, here we delete them from
         // groupingExpressions for condition resolving.
-        case a @ Aggregate(Seq(gs: BaseGroupingSets), _, _) =>
-          a.copy(groupingExpressions = gs.groupByExprs)
+        case a @ Aggregate(GroupingAnalytics(_, groupByExprs), _, _) =>
```

Comment:
can we pass `groupByExprs` as the parameter of `tryResolveHavingCondition`? It's already available in the caller side.

Reply:
Pass more parameter like current code can reduce more repeated code